### PR TITLE
UN-470: [BE] Record article promo component

### PR DESCRIPTION
**Pull request by ababic**
Ticket URL: https://national-archives.atlassian.net/browse/UN-470

## About these changes

Adds the `FeaturedRecordArticleBlock` block option, and makes it available as a content option on the `ArticlePage`

## How to check these changes

Edit an "Article Page" (e.g. [The Mangrove Nine](http://localhost:8000/admin/pages/87/edit/))

Within a content section, add a new block type and confirm you see the new "Featured record article" option. It should look a little bit like this:

Screenshot 2023-04-18 at 5 35 41 pm

The block just allows you to select a "Record Article", and that is all.

FE to be reviewed and refined in: https://national-archives.atlassian.net/browse/UN-169

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/blocks/featured_record_article.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{% load static wagtailimages_tags wagtailcore_tags %}
+
+{% if value.page.live or request.is_preview %}
```
Comment (William-Blackie) [inline templates/blocks/featured_record_article.html:3]: Thought: It's strange to handle this in the FE and could lead to a broken experience for editors that would be hard to figure out. I'm not suggestion that this is not the correct way but feel's fragile, is there anything we can do in the BE to enforce this?

**Review by William-Blackie — APPROVED** [review]

File: templates/blocks/featured_record_article.html (hunk 1 above)
Comment (ababic) [reply templates/blocks/featured_record_article.html:3]: Problem there is... it's perfectly valid for in-progress content to link to other in-progress content (maybe they are working on a 'section' of related content). Pages also change states in a way that is difficult to enforce, and probably shouldn't be enforced. If you linked to a live page, then that page is later unpublished temporarily while it undergoes an update, does that mean the page with the link is now invalid?

Comment (ababic) [reply templates/blocks/featured_record_article.html:3]: I can tell you from experience that 'validation' isn't the answer :)

Comment (ababic) [reply templates/blocks/featured_record_article.html:3]: There might be some way to do this sort of thing with custom chooser and structblocks in Wagtail, but that would definitely need to be a separate piece of work.

**Pull request merged**